akka-javasdk/src/main/java/akka/javasdk/JsonSupport.java

```diff
@@ -123,26 +131,81 @@ public static <T> Any encodeJson(T value, String jsonType) {
     }
   }
 
-  // FIXME do we really want all these to be public API?
+  /**
+   * @deprecated Use encodeToAkkaByteString
+   */
+  @Deprecated
   public static <T> ByteString encodeToBytes(T value) throws JsonProcessingException {
     return UnsafeByteOperations.unsafeWrap(
       objectMapper.writerFor(value.getClass()).writeValueAsBytes(value));
   }
 
-  public static <T> akka.util.ByteString encodeToAkkaByteString(T value) throws JsonProcessingException {
-    return akka.util.ByteString.fromArrayUnsafe(objectMapper.writerFor(value.getClass()).writeValueAsBytes(value));
+  /**
+   * Encode the given value as JSON using Jackson.
+   *
+   * @param value    the object to encode as JSON, must be an instance of a class properly annotated
+   *                 with the needed Jackson annotations.
+   * @throws IllegalArgumentException if the given value cannot be turned into JSON
+   */
+  public static <T> akka.util.ByteString encodeToAkkaByteString(T value) {
+    try {
+      return akka.util.ByteString.fromArrayUnsafe(objectMapper.writerFor(value.getClass()).writeValueAsBytes(value));
+    } catch (JsonProcessingException ex) {
+      throw new IllegalArgumentException(
+          "Could not encode [" + value.getClass().getName() + "] as JSON", ex);
+    }
   }
 
-  public static akka.util.ByteString encodeDynamicToAkkaByteString(String key, String value) throws JsonProcessingException {
-    ObjectNode dynamicJson = objectMapper.createObjectNode().put(key, value);
-    return akka.util.ByteString.fromArrayUnsafe(objectMapper.writeValueAsBytes(dynamicJson));
+  public static akka.util.ByteString encodeDynamicToAkkaByteString(String key, String value) {
+    try {
+      ObjectNode dynamicJson = objectMapper.createObjectNode().put(key, value);
+      return akka.util.ByteString.fromArrayUnsafe(objectMapper.writeValueAsBytes(dynamicJson));
+    } catch (JsonProcessingException ex) {
+      throw new IllegalArgumentException(
+          "Could not encode dynamic key/value as JSON", ex);
+    }
   }
 
-  public static akka.util.ByteString encodeDynamicCollectionToAkkaByteString(String key, Collection<?> values) throws JsonProcessingException {
-    ObjectNode objectNode = objectMapper.createObjectNode();
-    ArrayNode dynamicJson = objectNode.putArray(key);
-    values.forEach(v -> dynamicJson.add(v.toString()));
-    return akka.util.ByteString.fromArrayUnsafe(objectMapper.writeValueAsBytes(objectNode));
+  public static akka.util.ByteString encodeDynamicCollectionToAkkaByteString(String key, Collection<?> values) {
```

Comment:
Quite strange that this ever was public API I think it was mostly methods we needed ourselves internally that accidentally became public API. We should reconsider that and limit the surface to intentionally public API.

Reply:
deprecated those, and changed to internal JsonSerializer: https://github.com/akka/akka-sdk/pull/105/commits/c9f7b3baf36da4a9fa01b2a56ae456dd7ebff35c